### PR TITLE
CHK-463: Add index property to cancellation error

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Adds the `index` property to the error of cancelled tasks. This is useful for a scenario where you keep a separate data structure that is supposed to reflect the same state as the queue inside `OrderQueue`, such as the localStorage queue inside `order-items`. Without this we wouldn't be able to properly track the cancelled tasks position and sync the local queue with the in-memory queue.

#### How should this be manually tested?

[Workspace](https://batch--checkoutio.myvtex.com/).

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Review by jeffersontuc — APPROVED** [review]

File: react/modules/SequentialTaskQueue.ts [hunk 1]
```diff
@@ -153,78 +155,109 @@ export class SequentialTaskQueue {
...
+
+    if (queueIndex < 0) {
+      return queueIndex
+    }
+
+    // Add 1 here because the queue is always missing the first
+    // element, which is stored in `this.currentTask`, so we need
+    // to shift the index by 1.
+    return queueIndex + 1
+  }
```
Comment (wdsrocha) [inline react/modules/SequentialTaskQueue.ts:236]: Do you think it's worth to add an unit test for this?

Comment (lucasecdb) [reply react/modules/SequentialTaskQueue.ts:236]: sure! I'll fix the broken unit tests as well

**Review by wdsrocha — APPROVED** [review]
LGTM!

Btw, for future PRs, could you split the refactor or formatting into their own commit? I think this will make the PRs easier to review, as the reviewer will be able to focus only on the functional changes.

**Review by wdsrocha — DISMISSED** [review]

**Comment by lucasecdb** [comment]
> Btw, for future PRs, could you split the refactor or formatting into their own commit? I think this will make the PRs easier to review, as the reviewer will be able to focus only on the functional changes.

sorry about that 😅 I usually do this but I didn't pay attention to the changes I was commiting

**Review by jeffersontuc — APPROVED** [review]
Tests 😍

**Review by wdsrocha — DISMISSED** [review]

**Review by wdsrocha — APPROVED** [review]
:another_one:

**Review by jeffersontuc — APPROVED** [review]

**Pull request merged**